### ISSUE TITLE
bundle install: `rb_exc_raise': nested asn1 error (Segmentation fault) in Travis CI

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1627]
This doesn't happen on my Mac but I got this in Travis CI in Linux here: https://travis-ci.org/dry-rb/dry-initializer/jobs/508901190

Error

```
Fetching gem metadata from https://rubygems.org/..
Retrying fetcher due to error (2/4): OpenSSL::X509::CertificateError nested asn1 error.
Retrying fetcher due to error (3/4): OpenSSL::X509::CertificateError nested asn1 error.
Retrying fetcher due to error (4/4): OpenSSL::X509::CertificateError nested asn1 error.
/home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/truffle/truffle/cext.rb:1130:in `rb_exc_raise': nested asn1 error (OpenSSL::X509::CertificateError)
	from ruby.c:1989:in `rb_exc_raise'
	from ossl.c:300:in `ossl_raise'
	from ossl_x509cert.c:63:in `ossl_x509_new'
	from ossl_ssl.c:2091:in `ossl_ssl_get_peer_cert'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/truffle/truffle/cext_ruby.rb:41:in `peer_cert'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/openssl/ssl.rb:385:in `post_connection_check'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/net/http.rb:1016:in `connect'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/net/http.rb:936:in `do_start'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/net/http.rb:931:in `start'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/vendor/net-http-persistent/lib/net/http/persistent.rb:702:in `start'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/vendor/net-http-persistent/lib/net/http/persistent.rb:967:in `reset'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/vendor/net-http-persistent/lib/net/http/persistent.rb:630:in `connection_for'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/vendored_persistent.rb:23:in `connection_for'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/vendor/net-http-persistent/lib/net/http/persistent.rb:996:in `request'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/fetcher/downloader.rb:56:in `request'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/fetcher/downloader.rb:17:in `fetch'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/fetcher/compact_index.rb:117:in `call'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/compact_index_client/updater.rb:53:in `block in update'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/tmpdir.rb:93:in `mktmpdir'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/compact_index_client/updater.rb:31:in `update'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/compact_index_client.rb:82:in `update'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/compact_index_client.rb:98:in `update_info'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/compact_index_client.rb:55:in `block in dependencies'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/fetcher/compact_index.rb:88:in `call'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/fetcher/compact_index.rb:88:in `block (3 levels) in compact_index_client'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/worker.rb:64:in `call'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/worker.rb:64:in `apply_func'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/worker.rb:59:in `block in process_queue'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/worker.rb:56:in `loop'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/worker.rb:56:in `process_queue'
	from /home/travis/.rvm/rubies/truffleruby-1.0.0-rc14/lib/mri/bundler/worker.rb:90:in `block (2 levels) in create_threads'
/home/travis/.travis/functions: line 315:  8111 Segmentation fault      (core dumped) bundle install --without benchmarks tools --path=${BUNDLE_PATH:-vendor/bundle}
```

**Comment by deepj** [comment]
Again in case of dry-initializer https://travis-ci.org/dry-rb/dry-initializer/jobs/508921011

**Comment by deepj** [comment]
Here https://travis-ci.org/dry-rb/dry-struct/jobs/508921345

**Comment by deepj** [comment]
@eregon @chrisseaton this would have a high priority because it's failing in Travis CI

**Comment by aardvark179** [comment]
From the trace it looks like a malloc failure. It might be possible to reproduce in a docker container.

**Comment by deepj** [comment]
Again here https://travis-ci.org/dry-rb/dry-core/jobs/509683061

**Comment by eregon** [comment]
@deepj This last one seems a different error (Segmentation fault/Bus error), could you report it separately?

**Comment by nirvdrum** [comment]
I'm sorry for the slow update. This was addressed in f08dd1ac65f22da9bb976ee8792429bfce6d04c5, which updates the post-install hook to always recompile OpenSSL for the system it's running on. The fix will be available in the next release.

If you're still seeing the issue in the next release, please comment here and we can re-open the issue.
